### PR TITLE
Feat (number) / inRange function

### PR DESCRIPTION
**Pull request by mertdy**
## Description

The equivalent of lodash's inRange function. <a href="https://lodash.com/docs/4.17.15#inRange" target="_blank">lodash.inRange</a>

## Checklist

- [x] Changes are covered by tests if behavior has been changed or added
- [x] Tests have 100% coverage
- [ ] If code changes were made, the version in `package.json` has been bumped (matching semver)
- [x] If code changes were made, the `yarn build` command has been run and to update the `cdn` directory
- [x] If code changes were made, the documentation (in the `/docs` directory) has been updated

## Resolves

None. It is a new feature.

### CODE REVIEW

File: src/number.ts [hunk 1]
```diff
@@ -1,3 +1,38 @@
...
+ * @param {number} number The number to check.
+ * @param {number} start The start of the range. Inclusive.
+ * @param {number} end The end of the range. Exclusive.
+ * @returns {boolean} Returns `true` if `number` is in the range, else `false`.
+ */
+export const inRange = (
+  number: number,
+  start: number,
+  end?: number
+): boolean => {
```
Comment (rayepps) [inline src/number.ts:18]: In the interest of great DX I would make this a `function` with two overloads: one with `(number, end)` and one with `(number, start, end)`. Small difference but will give developers a little better typeahead/Intellisense when looking at the arguments.

**Review by rayepps — CHANGES_REQUESTED** [review]
Hey @mertdy sorry for the very long delay here. I love this PR, I love the docs, the build, the format. You're making it easy on me.

I just had one small comment. If you're still interested in getting this in we can release asap.

File: src/number.ts (hunk 1 above)
Comment (mertdy) [reply src/number.ts]: Done 👍🏻

**Comment by mertdy** [comment]
Hey @rayepps, I added overloading functions. If you have time to review, we can proceed the PR.

**Comment by rayepps** [comment]
Thank you @mertdy finally merged it 🙃 🎉 will go out in next release of v12.0.0 (likely today)

**Pull request merged**